ctru-rs/src/services/ndsp/mod.rs

```diff
@@ -209,13 +276,35 @@ impl AudioFormat {
     /// 16 bit formats return 2 (bytes)
     pub fn sample_size(self) -> u8 {
         match self {
-            AudioFormat::PCM8Mono | AudioFormat::ADPCMMono => 1,
+            AudioFormat::PCM8Mono => 1,
             AudioFormat::PCM16Mono | AudioFormat::PCM8Stereo => 2,
             AudioFormat::PCM16Stereo => 4,
         }
     }
 }
 
+impl fmt::Display for NdspError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::InvalidChannel(id) => write!(f, "Audio Channel with id {id} doesn't exist. Valid channels have an id between 0 and 23."),
+            Self::ChannelAlreadyInUse(id) => write!(f, "Audio Channel with id {id} is already being used. Drop the other instance if you want to use it here."),
+            Self::WaveAlreadyQueued(id) => write!(f, "The selected WaveInfo is already playing on channel {id}.")
+        }
+    }
+}
+
+impl error::Error for NdspError {
+    fn description(&self) -> &str {
```

Comment:
Any reason to define this when already using `Display`? It looks [to be deprecated](https://doc.rust-lang.org/std/error/trait.Error.html#method.description) for a while